src/plaid/storage/common/reader.py

```diff
@@ -0,0 +1,203 @@
+# -*- coding: utf-8 -*-
+#
+# This file is subject to the terms and conditions defined in
+# file 'LICENSE.txt', which is part of this source code package.
+#
+#
+
+"""Common storage reader utilities.
+
+This module provides common utilities for reading dataset metadata, problem definitions,
+and other auxiliary files from disk or downloading them from Hugging Face Hub.
+"""
+
+import pickle
+import tempfile
+from pathlib import Path
+from typing import Any, Optional, Union
+
+import yaml
+from huggingface_hub import hf_hub_download, snapshot_download
+
+from plaid import ProblemDefinition
+
+# ------------------------------------------------------
+# Load from disk
+# ------------------------------------------------------
+
+
+def load_infos_from_disk(path: Union[str, Path]) -> dict[str, Any]:
+    """Load dataset information from a YAML file stored on disk.
+
+    Args:
+        path (Union[str, Path]): Directory path containing the `infos.yaml` file.
+
+    Returns:
+        dict[str, dict[str, str]]: Dictionary containing dataset infos.
+    """
+    infos_fname = Path(path) / "infos.yaml"
+    with infos_fname.open("r") as file:
+        infos = yaml.safe_load(file)
+    return infos
+
+
+def load_problem_definitions_from_disk(
+    path: Union[str, Path],
+) -> Optional[list[ProblemDefinition]]:
+    """Load ProblemDefinitions from disk.
+
+    Args:
+        path (Union[str, Path]): The directory path for loading.
+
+    Returns:
+        Optional[list[ProblemDefinition]]: List of loaded problem definitions, or None if not found.
+    """
+    pb_def_dir = Path(path) / Path("problem_definitions")
+
+    if pb_def_dir.is_dir():
+        pb_defs = []
+        for p in pb_def_dir.iterdir():
+            if p.is_file():
+                pb_def = ProblemDefinition()
+                pb_def._load_from_file_(pb_def_dir / Path(p.name))
+                pb_defs.append(pb_def)
+        return pb_defs
+    else:
+        return None  # pragma: no cover
```

Comment:
Should we print something to tell the user that no problem definition was found ?